src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -45,13 +49,15 @@ export function DiscoverMainRoute() {
     data,
     toastNotifications,
     http: { basePath },
+    dataViewEditor,
   } = services;
   const [error, setError] = useState<Error>();
   const [savedSearch, setSavedSearch] = useState<SavedSearch>();
   const indexPattern = savedSearch?.searchSource?.getField('index');
   const [indexPatternList, setIndexPatternList] = useState<Array<SavedObject<DataViewAttributes>>>(
     []
   );
+  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
```

Comment:
Small suggestion: would be great to avoid jumping from "Welcome to Analytics" to No data view components and show a loading instead.

![May-17-2022 12-06-28](https://user-images.githubusercontent.com/1415710/168795604-1fbe739d-a3cc-4e8d-893c-bb5a98893489.gif)

Reply:
Yes, that's already being tackled in the component itself: https://github.com/elastic/kibana/pull/132272